### ISSUE TITLE
embedded scene in PNG/SVG should reuse fileHandle

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#2261]
When opening a scene from a PNG/SVG file, we should retain the `fileHandle` and on save (via save button or `Cmd+S`) we should write back to it: both export latest PNG and embed latest scene.

First suggested in https://github.com/excalidraw/excalidraw/issues/2250 @pomdtr

Considerations: 

- doing this, we'll overload the save behavior to now either save JSON, or export PNG+JSON depending on context. This may/may not be a UX problem.

**Comment by alsuren** [comment]
I created https://github.com/excalidraw/excalidraw/pull/3516 as a proof-of-concept that this is possible.

I don't know how to handle the Save As use-case, but I've sketched up my thoughts about it in the PR: https://github.com/excalidraw/excalidraw/pull/3516/files#diff-6731ff2f1f238d489cf0373e9d5ae1d81b523c608b571291a14401526d84e1b4R96-R104

It would be good to have a steer on how this should be handled from a UX perspective, or from someone who is more familiar with what's possible to do with filesystems on the web. It's probably reasonable to punt this problem into a separate ticket.
